interface/src/components/upload2.js

```diff
@@ -47,16 +48,21 @@ class Upload extends Component{
                console.log(key);
               await parseToCsv(this.state[key].url)
                .then((data)=>{
-                   this.setState({[key]:data})
+                    str.pop();
+                    this.setState({[key]:data})
                })
-               .catch(err=> console.log(err))
+               .catch(err=> err_link.push(key))
            }
        }
        if(str.length===0){
            this.props.getFiles(this.state)
-           return this.props.history.push("/visual");
+           return this.props.history.push("/validate");
        }
-       str.join(", ");
+       if(err_link.length!==0){
+            err_link=err_link.join(", ")
+            alert(`Please check your google sheet link for ${err_link} files`);
+       }
+       str= str.join(", ");
        alert(`You have not uploaded your ${str} files`);
```

Comment:
Same comments as in `upload1.js`

Reply:
Actually, there is a component for errors ([interface/src/components/error.js](https://github.com/PecanProject/BETYdb-YABA/blob/688fcfe3f7e08535013586c06d0689ccb0f4d114/interface/src/components/error.js)) but we just want to overlay a message for an invalid API key rather redirecting to an error page